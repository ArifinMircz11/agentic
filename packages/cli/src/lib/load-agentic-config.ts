import { parseZodSchema } from '@agentic/platform-core'
import {
  type AgenticProjectConfigOutput,
  agenticProjectConfigSchema
} from '@agentic/platform-schemas'
import { loadConfig } from 'unconfig'

export async function loadAgenticConfig({
  cwd
}: {
  cwd?: string
}): Promise<AgenticProjectConfigOutput> {
  const { config } = await loadConfig({
    cwd,
    sources: [
      {
        files: 'agentic.config',
        extensions: ['ts', 'mts', 'cts', 'js', 'mjs', 'cjs', 'json']
      }
    ]
  })

  return parseZodSchema(agenticProjectConfigSchema, config)
}
