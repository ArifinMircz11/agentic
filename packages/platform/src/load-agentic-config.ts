import type { AgenticProjectConfig } from '@agentic/platform-types'
import { loadConfig } from 'unconfig'

import { validateAgenticProjectConfig } from './validate-agentic-project-config'

export async function loadAgenticConfig({
  cwd
}: {
  cwd?: string
} = {}): Promise<AgenticProjectConfig> {
  const { config } = await loadConfig({
    cwd,
    sources: [
      {
        files: 'agentic.config',
        extensions: ['ts', 'mts', 'cts', 'js', 'mjs', 'cjs', 'json']
      }
    ]
  })

  return validateAgenticProjectConfig(config)
}
