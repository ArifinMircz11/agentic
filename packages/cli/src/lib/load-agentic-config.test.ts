import path from 'node:path'
import { fileURLToPath } from 'node:url'

import { describe, expect, test } from 'vitest'

import { loadAgenticConfig } from './load-agentic-config'

const fixtures = [
  'basic-raw-free-ts',
  'basic-raw-free-json',
  'pricing-freemium',
  'pricing-pay-as-you-go',
  'pricing-3-plans',
  'pricing-monthly-annual',
  'pricing-custom-0'
]

const invalidFixtures = [
  'pricing-base-inconsistent',
  'pricing-custom-inconsistent',
  'pricing-empty-0',
  'pricing-empty-1',
  'pricing-empty-2',
  'pricing-duplicate-0',
  'pricing-duplicate-1',
  'invalid-origin-url-0',
  'invalid-origin-url-1',
  'invalid-origin-url-2',
  'invalid-origin-url-3',
  'invalid-name-0',
  'invalid-name-1',
  'invalid-name-2',
  'invalid-name-3',
  'invalid-name-4'
]

const fixturesDir = path.join(
  fileURLToPath(import.meta.url),
  '..',
  '..',
  '..',
  'fixtures'
)

const invalidFixturesDir = path.join(fixturesDir, 'invalid')

describe('loadAgenticConfig', () => {
  for (const fixture of fixtures) {
    test(
      `${fixture}`,
      {
        timeout: 60_000
      },
      async () => {
        const fixtureDir = path.join(fixturesDir, fixture)

        const config = await loadAgenticConfig({ cwd: fixtureDir })
        expect(config).toMatchSnapshot()
      }
    )
  }

  for (const fixture of invalidFixtures) {
    test(
      `invalid: ${fixture}`,
      {
        timeout: 60_000
      },
      async () => {
        const fixtureDir = path.join(invalidFixturesDir, fixture)

        await expect(
          loadAgenticConfig({ cwd: fixtureDir })
        ).rejects.toThrowErrorMatchingSnapshot()
      }
    )
  }
})
