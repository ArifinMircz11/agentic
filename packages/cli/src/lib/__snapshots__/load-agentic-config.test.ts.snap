// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loadAgenticConfig > basic-raw-free-json 1`] = `
{
  "name": "My Project",
  "originAdapter": {
    "location": "external",
    "type": "raw",
  },
  "originUrl": "https://jsonplaceholder.typicode.com",
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "slug": "free",
    },
  ],
}
`;

exports[`loadAgenticConfig > basic-raw-free-ts 1`] = `
{
  "name": "My Project",
  "originAdapter": {
    "location": "external",
    "type": "raw",
  },
  "originUrl": "https://jsonplaceholder.typicode.com",
  "pricingPlans": [
    {
      "lineItems": [
        {
          "amount": 0,
          "slug": "base",
          "usageType": "licensed",
        },
      ],
      "name": "Free",
      "slug": "free",
    },
  ],
}
`;
